client/my-sites/themes/themes-selection.jsx

```diff
@@ -212,13 +202,21 @@ class ThemesSelection extends Component {
 				} else {
 					defaultOption = options.activate;
 				}
-				this.props.setThemePreviewOptions( themeId, defaultOption, secondaryOption, null );
+				this.props.setThemePreviewOptions(
+					themeId,
+					defaultOption,
+					secondaryOption,
+					styleVariation
+				);
 				return action( t, context );
 			};
 		};
 
-		if ( options && options.preview ) {
-			options.preview.action = wrappedPreviewAction( options.preview.action );
+		if ( options ) {
+			options = addStyleVariation( options, styleVariation );
```

Comment:
Since we wrapped all `getUrl` functions with the selected style variation, how about passing the style variation to the 3rd argument to the `getUrl` function to let those functions decide how to handle the style variation? Does it make sense?

For example, the checkout URL might be incorrect as it should append the style variation to the `redirect_url` instead of appending it to the end of the URL directly

Reply:
Good catch! I've updated the code so that each option's `getUrl` is responsible for handling the style variation.